homeassistant/components/enocean/cover.py

```diff
@@ -0,0 +1,267 @@
+            self._is_closing = False
+        elif kwargs[ATTR_POSITION] > self._position:
+            self._is_opening = True
+            self._is_closing = False
+        elif kwargs[ATTR_POSITION] < self._position:
+            self._is_opening = False
+            self._is_closing = True
+
+        self.start_or_feed_watchdog()
+
+        telegram = [0xD2, 100 - kwargs[ATTR_POSITION], 0, 0, 1]
+        telegram.extend(self._sender_id)
+        telegram.extend([0x00])
+        self.send_telegram(telegram)
+
+    def stop_cover(self, **kwargs) -> None:
+        """Stop any cover movement."""
+        self.stop_watchdog()
+        self._state_changed_by_command = True
+        self._is_opening = False
+        self._is_closing = False
+
+        telegram = [0xD2, 2]
+        telegram.extend(self._sender_id)
+        telegram.extend([0x00])
+        self.send_telegram(telegram)
+
+    def value_changed(self, packet):
+        """Fire an event with the data that have changed.
+
+        This method is called when there is an incoming packet associated
+        with this platform.
+        """
+        # position is inversed in Home Assistant and in EnOcean:
+        # 0 means 'closed' in Home Assistant and 'open' in EnOcean
+        # 100 means 'open' in Home Assistant and 'closed' in EnOcean
+
+        new_position = 100 - packet.data[1]
+
+        if self._position is not None:
+            if self._state_changed_by_command:
+                self._state_changed_by_command = False
+            elif new_position in (0, 100, self._position):
+                self._is_opening = False
+                self._is_closing = False
+                self.stop_watchdog()
+            elif new_position > self._position:
+                self._is_opening = True
+                self._is_closing = False
+                self.start_or_feed_watchdog()
+            elif new_position < self._position:
+                self._is_opening = False
+                self._is_closing = True
+                self.start_or_feed_watchdog()
+
+        self._position = new_position
+        if self._position == 0:
+            self._is_closed = True
+        else:
+            self._is_closed = False
+
+        self.schedule_update_ha_state()
+
+    def send_telegram(self, data):
+        """Send a telegram via the EnOcean dongle to only this device."""
+        # optional data contains: number of subtelegrams (fixed to 3 for sending),
+        # destination id, max dBm (0xFF) for sending and security level 0
+        packet = Packet(0x01, data=data, optional=[3] + self.dev_id + [0xFF, 0])
+        dispatcher_send(self.hass, SIGNAL_SEND_MESSAGE, packet)
+
+    def start_or_feed_watchdog(self):
+        """Start or feed the 'movement stop' watchdog."""
+        self._watchdog_seconds_remaining = self._watchdog_timeout
+
+        if self._watchdog_enabled:
+            return
+
+        self._watchdog_enabled = True
+        self._hass.async_create_task(self.watchdog())
```

Comment:
We're not allowed to call the async api from sync context.
```suggestion
        self._hass.create_task(self.watchdog())
```